proposals/roles.md

```diff
@@ -0,0 +1,155 @@
+﻿# Roles
+
+This proposal is divided into three parts, all relating to extending existing types:
+A. static extensions
+B. roles and extensions with members
+C. roles and extensions that implement interfaces
+
+The syntax for a roles is as follows:
+
+```antr
+type_declaration
+	: role_declaration // add
+	| extension_declaration // add
+	| ...
+	;
+
+role_declaration
+    : modifier* 'role' identifier type_parameter_list? role_underlying_type type_parameter_constraints_clause* role_body
+    ;
+
+extension_declaration
+    : modifier* 'extension' identifier type_parameter_list? role_underlying_type type_parameter_constraints_clause* role_body
+    ;
```

Comment:
@MadsTorgersen Continued feedback that i think that role/extension are not good terms.